lib/job.js

```diff
@@ -27,7 +27,13 @@ class Job extends Emitter {
           client.hget(queue.toKey('jobs'), jobId, done)
         )
       )
-      .then((data) => (data ? Job.fromData(queue, jobId, data) : null));
+      .then((data) =>
+        data
+          ? Job.fromData(queue, jobId, data)
+          : queue.settings.clearMissingJobs
```

Comment:
I could also imagine some pluggable handler for this case, but that's out of scope.